### ISSUE TITLE
GameObj auto-delete (via X/YSpeed) can cause objects to get unexpectedly deleted

### ISSUE THREAD

**Issue by davecparker** [davecparker/code12#328]
(Possible design flaw, although working as documented)
For example, if a game is tracking a single bullet object and resetting it when re-fired, the bullet may get auto-deleted and mysteriously go away.

**Comment by davecparker** [comment]
Possibly even more surprising is that this happens even if the object has .visible = false

**Comment by davecparker** [comment]
I decided to fix this because it can hurt beginning uses before the user knows enough to work around it. I ran into it when trying to make a lower level sample. 

The fix is not perfect, but should do the right thing in practice in most cases. Now the rule (heuristic) for auto-deletion of objects is:

Lua runtime: If more than 100 objects have been created since start() completed (then clearly the program is creating them dynamically), then it auto-deletes objects using X/YSpeed that are more than 100 units off-screen, but only ones after the 50th in the stacking order (to try to avoid doing anything with core objects created early, no matter what).  Nothing is deleted for an app having < 100 objects no matter what.

Java runtime: If the object count is > 500 then it auto-deletes objects using X/YSpeed that are more than 100 units off-screen.
